backend/src/main/scala/cromwell/backend/ReadLikeFunctions.scala

```diff
@@ -2,13 +2,19 @@ package cromwell.backend
 
 import cromwell.core.io.AsyncIoFunctions
 import cromwell.core.path.PathFactory
+import cromwell.filesystems.drs.{DrsPath, DrsResolver}
 import wom.expression.IoFunctionSet
 
 import scala.concurrent.Future
 import scala.util.Try
 
 trait ReadLikeFunctions extends PathFactory with IoFunctionSet with AsyncIoFunctions {
 
+  override def resolvedPath(value: String): Future[String] = buildPath(value) match {
+    case drsPath: DrsPath => DrsResolver.getSimpleGsUri(drsPath).unsafeToFuture().map(_.getOrElse(value))
```

Comment:
Are you making this a Future because getSimpleGsUri is asynchronous? I tried looking through the methods, but didn't see anything obviously async.

Reply:
Yes, `DrsResolver.getSimpleGsUri(drsPath)` is asynchronous. It has to go out to a remote DRS server and resolve a DRS path into a cloud path.

Reply:
At a high level, resolving a DRS URI has to call Martha, so it is an asynchronous operation. `DrsResolver`/`DrsPathResolver` use cats-effect (`IO`) instead of `Future` to enclose the side-effecty code. The main difference here is that `IO` defers execution until explicitly evaluated (via `unsafeRunAsync`, `unsafeToFuture`, etc.) whereas `Future` begins execution immediately when it is instantiated.

My question is... why drop into a `Future` world here instead of carrying the `IO` forward? Maybe it's just to be consistent with other methods in this class?

Reply:
Everything in this API returns a `Future[_]` rather than `IO[_]`. I suspect that has more to do with the baseline code having been written in an era before Cromwell took up `IO` rather than a philosophical preference.

Reply:
Using `getSimpleGsUri` might be creating more work than necessary. The thing this is returning is built from pieces returned from Martha (I think) and, based on what I see in `DrsResolver`, also has some conditions that I don't understand. I think what we want here is just whatever Martha returns for `fileName`... though I'm not 100% sure; maybe those extra conditions are important in this context too. Anyway, if we just take `fileName`, we won't have to split on `/` later.